### PR TITLE
Closes #138

### LINKED ISSUE #138: Create dataset loader for PharmaCoNER
## Adding a Dataset
- **Name:** *PharmaCoNER*
- **Description:** *Following the outline of previous chemical and drug NER efforts, in particular the BioCreative CHEMDNER tracks, we organize the first task on chemical and drug mention recognition from Spanish medical texts, namely from a corpus of Spanish clinical case studies.*
- **Task:** *NER, NED*
- **Paper:** *[PharmaCoNER: Pharmacological Substances, Compounds and proteins Named Entity Recognition track](https://aclanthology.org/D19-5701/)*
- **Data:** *[Challenge Homepage](https://temu.bsc.es/pharmaconer/index.php/datasets/)*
- **License:** *[CC BY 4.0](https://creativecommons.org/licenses/by/4.0/)*
- **Motivation:** *Spanish langauge dataset*

### PR DESCRIPTION
**Pull request by mapama247**
- **Name:** *PharmaCoNER*
- **Description:** *Manually annotated collection of clinical case studies from Spanish medical publications.*
- **Paper:** *[link to the dataset paper](https://aclanthology.org/D19-5701.pdf)*
- **Data:** *[link to the online home of the dataset](https://temu.bsc.es/pharmaconer/index.php/datasets/)*

### Checkbox

- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `biodatasets/pharmaconer/pharmaconer.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_BIGBIO_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `BigBioConfig` for the source schema and one for a bigbio schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_bigbio biodatasets/pharmaconer/pharmaconer.py`.
- [x] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

File: biodatasets/pharmaconer/pharmaconer.py [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+_HOMEPAGE = "https://temu.bsc.es/pharmaconer/index.php/datasets/"
+
+_LICENSE = "Creative Commons Attribution 4.0 International"
+
+_URLS = {
+    "pharmaconer": "https://zenodo.org/record/4270158/files/pharmaconer.zip?download=1",
+}
+
+_SUPPORTED_TASKS = [Tasks.NAMED_ENTITY_RECOGNITION]
```
Comment (sg-wbi) [inline biodatasets/pharmaconer/pharmaconer.py:83]: Need to add ```Tasks.NAMED_ENTITY_DISAMBIGUATION``` for `subtrack 2`

File: biodatasets/pharmaconer/pharmaconer.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+        """
+        Downloads/extracts the data to generate the train, validation and test splits.
+
+        Each split is created by instantiating a `datasets.SplitGenerator`, which will
+        call `this._generate_examples` with the keyword arguments in `gen_kwargs`.
+        """
+
+        data_dir = dl_manager.download_and_extract(_URLS["pharmaconer"])
+
+        return [
```
Comment (sg-wbi) [inline biodatasets/pharmaconer/pharmaconer.py:208]: Here you should add the data for `subtrack 2`, which contains normalization (concept indexing ) information. See [here](https://temu.bsc.es/pharmaconer/) for details

**Review by sg-wbi — COMMENTED** [review]
@mapama247 Thank you very much for your contribution! Everything LGTM, however the dataset script it's not complete, in the sense that it offers `Tasks.NAMED_ENTITY_DISAMBIGUATION` as well and not only `Tasks.NAMED_ENTITY_RECOGNITION`. Could you please check my comments? Thank you!

**Comment by sg-wbi** [comment]
Good point. Sorry I misunderstood the data. What I would do:
```
_SUPPORTED_TASKS = [Tasks.NAMED_ENTITY_RECOGNITION, Tasks.TEXT_CLASSIFICATION] 
```
With `subset_id  == subtrack 1` you have  NAMED_ENTITY_RECOGNITION, while with `subset_id == subtrack 2` you have TEXT_CLASSIFICATION. Does it make sense?

You can have a look [here for multiple `subset_id`s](https://github.com/bigscience-workshop/biomedical/blob/master/examples/bioasq.py#L401).

**Comment by mapama247** [comment]
hi again @sg-wbi , I have updated the script so that it includes the text classification task as you suggested. Let me know if it is okay now!

**Comment by sg-wbi** [comment]
Thank you adding a new view @mapama247! This makes the dataloader complete. One thing though: I checked an example output of the dataset and something seems to be wrong.
```python
[ins] In [15]: from datasets import load_dataset
[ins] In [16]: ds["train"][54]
Out[16]: 
{'id': '54',
 'document_id': 'S0004-06142007000600016-2',
 'text': 'Paciente varón de 35 años con tumoración en polo superior de teste derecho hallada de manera casual durante una autoexploración, motivo por el cual acude a consulta de urología donde se realiza exploración física, apreciando masa de 1cm aproximado de diámetro dependiente de epidídimo, y ecografía testicular, que se informa como lesión nodular sólida en cabeza de epidídimo derecho. Se realiza RMN. Confirmando masa nodular, siendo el tumor adenomatoide de epidídimo la primera posibilidad diagnóstica.\n\nSe decide, en los dos casos, resección quirúrgica de tumoración nodular en cola epidídimo derecho, sin realización de orquiectomía posterior.\nEn ambos casos se realizó examen anátomopatológico de la pieza quirúrgica. Hallazgos histológicos macroscópicos: formación nodular de 1,5 cms (caso1) y 1,2 cms (caso 2) de consistencia firme, coloración blanquecina y bien delimitada. Microscópicamente se observa proliferación tumoral constituida por estructuras tubulares en las que la celularidad muestra núcleos redondeados y elongados sin atipia citológica y que ocasionalmente muestra citoplasmas vacuolados, todo ello compatible con tumor adenomatoide de epidídimo.\n\n',
 'labels': ['372776000', '39290007', '372694001']}
```
In this example I would assume that if I look in `train-set_1.1/train/subtrack2/S0004-06142007000600016-2.tsv ` I would expect to find `['372776000', '39290007', '372694001']`. However, if I check that file it's empty:
```bash
$ wc -l train-set_1.1/train/subtrack2/S0004-06142007000600016-2.tsv
0 train-set_1.1/train/subtrack2/S0004-06142007000600016-2.tsv
```
Could you please check that the examples are loaded correctly?

**Review by sg-wbi — APPROVED** [review]
@mapama247 Thank you for fixing the script. This is good to go now! Thank you for the contribution!

**Pull request merged**